kasa/smart/smartmodule.py

```diff
@@ -93,3 +93,8 @@ def data(self):
     def supported_version(self) -> int:
         """Return version supported by the device."""
         return self._device._components[self.REQUIRED_COMPONENT]
+
+    @property
+    def is_available(self):
+        """Is the module available on this device."""
+        return True
```

Comment:
I'm wondering about the naming of the property, as the current logic only works if the data is available in the initial queries, right?

Probably also makes sense to describe that in the docstring and/or mark this as private as it is not something downstreams should be concerned about.

Reply:
Renamed `_check_supported` and made an async function in case we ever need to make an api call to check supported in the future.